components/brave_new_tab_ui/components/search/config.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+import { SearchEngineInfo } from "../../api/background";
+
+// At some point we might want to store this in prefs, but
+// at this point its only used on Desktop, and only on the NTP
+// so this seems fine.
+const ENABLED_SEARCH_ENGINES_KEY = 'search-engines'
+const LAST_SEARCH_ENGINE_KEY = 'last-search-engine'
+
+const braveSearchOrigin = 'https://search.brave.com'
+
+let cache: Record<string, boolean> | undefined
+
+const getConfig = () => {
+  if (!cache) {
+    cache = JSON.parse(localStorage.getItem(ENABLED_SEARCH_ENGINES_KEY)!) ?? {
+      // Default to enabling Brave Search
+      [braveSearchOrigin]: true
```

Comment:
I'm curious we're intentionally use brave search as default engine, regardless of the omnibox's default engine. This would be only applied to the realbox so I don't have any objection to it, just curiosity.

Reply:
I have no strong preference, but should we `setDefaultEngine` with brave search, when there's nothing recorded before?

Reply:
Aha! Thanks.